core/services/standardcapabilities/delegate.go

```diff
@@ -111,6 +117,20 @@ func (d *Delegate) ServicesForSpec(ctx context.Context, spec job.Job) ([]job.Ser
 	log := d.logger.Named("StandardCapabilities").Named(spec.StandardCapabilitiesSpec.GetID())
 
 	kvStore := job.NewKVStore(spec.ID, d.ds)
+
+	var accountIDs []string
+	var signers []crypto.Signer
+	if d.ks.P2P() != nil && d.externalPeerWrapper != nil {
+		key, err := d.ks.P2P().GetOrFirst(p2pkey.PeerID(d.externalPeerWrapper.GetPeer().ID()))
+		if err != nil {
+			log.Warnw("Failed to get P2P key", "error", err, "peerID", d.externalPeerWrapper.GetPeer().ID())
+		} else {
+			accountIDs = append(accountIDs, "P2P_SIGNER")
+			signers = append(signers, key)
+		}
+	}
+	keystore := core.NewMultiAccountSigner(accountIDs, signers)
```

Comment:
@vreff Can you explain more what your intention is here? Do you need access to multiple keys?